packages/alfa-rules/src/sia-r14/rule.ts

```diff
@@ -152,6 +150,11 @@ function childrenPerceivableText(node: Node, device: Device): string {
   return result;
 }
 
+// Includes removing all punctiation (underscore, hypen, brackets, quotation marks, etc)
+function normalize(input: string): string {
+  return input.trim().toLowerCase().replace(/\p{P}/gu, "").replace(/\s+/g, " ");
+}
+
```

Comment:
I think it might be better to keep the common `normalise` and use here `normalise(foo).replace(/\p{P}/gu, "")`.
That way we'll avoid having several normalisations and keep the shared one with a single meaning. Then callers may do additional treatments.

Also great find with the `\p` magic 👍

Reply:
We can do that, I'm suspicious that it might turn into `normalise(foo).replace(/\p{P}/gu, "").replace(/\s+/g, " ")` though to handle `Hello? - ( world )`

Reply:
Ah, yes…
Then `normalise(foo.replace(/\p{P}/gu, "")` ?

Reply:
It will look like this within the expectation:

```Javascript
const perceivableInnerTextFromElement =
  getPerceivableInnerTextFromElement(target, device);
const textContent = normalize(
  // Removes all punctiation (underscore, hypen, brackets, quotation marks, etc)
  perceivableInnerTextFromElement.replace(/\p{P}/gu, "")
);
let name = "";

const accessibleNameIncludesTextContent = test(
  hasAccessibleName(device, (accessibleName) => {
    const value = accessibleName.value;
    // Removes all punctiation (underscore, hypen, brackets, quotation marks, etc)
    name = normalize(value.replace(/\p{P}/gu, ""));
    return name.includes(textContent);
  }),
  target
);
```

Should I leave the comments at both places? I'll push it up if this looks good :)

Reply:
Maybe make a local function so that we don't duplicate code (I do foresee we might want to add some other processing at some point…)